internal/pkg/template/app.go

```diff
@@ -69,8 +89,10 @@ func (t *Template) ParseApp(name string, data interface{}, options ...ParseOptio
 func withAppParsingFuncs() ParseOption {
 	return func(t *template.Template) *template.Template {
 		return t.Funcs(map[string]interface{}{
```

Comment:
Could you help me understand how this works, and why the map is necessary versus a struct?

Reply:
Sure! so while parsing go templates you can specify some methods to do transformations (https://golang.org/pkg/text/template/#Template.Funcs). For example, i can write in a template `{{toSnakeCase "efeIsHorrible"}}` and it will become `EFE_IS_HORRIBLE` when the template is rendered.

The input that they require is a [FuncMap](https://golang.org/pkg/text/template/#FuncMap) (map[string]interface{}) so it can't be a struct that we define.

Reply:
Very cool, thanks for the clear explanation!